### ISSUE TITLE
compile deps `certifi` for error

### ISSUE THREAD

**Issue by hpyhacking** [benoitc/hackney#261]
Use new version compile error, fixed when force using 1.4.4 version.

```
Erlang/OTP 18
Elixir 1.1
```

```
~/test mix new test
...... # just add deps {:hackney, "~> 1.4"} in mix.exs

~/test mix deps.get
Running dependency resolution
Dependency resolution completed successfully
  certifi: v0.3.0
  hackney: v1.4.5
  idna: v1.0.2
  mimerl: v1.0.0
  ssl_verify_hostname: v1.0.5
* Getting hackney (Hex package)
Checking package (http://hex.peatio.com/tarballs/hackney-1.4.5.tar)
Using locally cached package
Unpacked package tarball (/Users/hpyhacking/.hex/packages/hackney-1.4.5.tar)
* Getting ssl_verify_hostname (Hex package)
Checking package (http://hex.peatio.com/tarballs/ssl_verify_hostname-1.0.5.tar)
Using locally cached package
Unpacked package tarball (/Users/hpyhacking/.hex/packages/ssl_verify_hostname-1.0.5.tar)
* Getting mimerl (Hex package)
Checking package (http://hex.peatio.com/tarballs/mimerl-1.0.0.tar)
Using locally cached package
Unpacked package tarball (/Users/hpyhacking/.hex/packages/mimerl-1.0.0.tar)
* Getting idna (Hex package)
Checking package (http://hex.peatio.com/tarballs/idna-1.0.2.tar)
Using locally cached package
Unpacked package tarball (/Users/hpyhacking/.hex/packages/idna-1.0.2.tar)
* Getting certifi (Hex package)
Checking package (http://hex.peatio.com/tarballs/certifi-0.3.0.tar)
Using locally cached package
Unpacked package tarball (/Users/hpyhacking/.hex/packages/certifi-0.3.0.tar)
~/test mix
==> idna (compile)
Compiled src/idna_unicode.erl
Compiled src/punycode.erl
Compiled src/idna.erl
Compiled src/idna_ucs.erl
Compiled src/idna_unicode_data.erl
WARN:  Missing plugins: [rebar3_hex]
==> mimerl (compile)
Compiled src/mimerl.erl
==> ssl_verify_hostname (compile)
Compiled src/ssl_verify_hostname.erl
==> certifi (compile)
Compiled src/certifi.erl
Compiled src/certifi_weak.erl
Compiled src/certifi_cacerts.erl
WARN:  Expected /Users/hpyhacking/test/_build/dev/lib/cowboy to be an app dir (containing ebin/*.app), but no .app found.
==> hackney (compile)
WARN:  Expected /Users/hpyhacking/test/_build/dev/lib/cowboy to be an app dir (containing ebin/*.app), but no .app found.
Dependency not available: cowboy-.* ({git,
                                      "https://github.com/ninenines/cowboy.git",
                                      {tag,"1.0.4"}})
ERROR: compile failed while processing /Users/hpyhacking/test/deps/hackney: rebar_abort
** (Mix) Could not compile dependency :hackney, "/Users/hpyhacking/.mix/rebar" command failed. You can recompile this dependency with "mix deps.compile hackney", update it with "mix deps.update hackney" or clean it with "mix deps.clean hackney"
```

**Comment by benoitc** [comment]
should be solved in latest master. Can you give it a try?

**Comment by curtislinden** [comment]
on master

urtis:blocksworld-play-site/ (master✗) $ mix deps.compile hackney                                                                                           [10:29:10]
WARN:  Expected /Users/curtis/dev/blocksworld-play-site/_build/dev/lib/mimerl to be an app dir (containing ebin/*.app), but no .app found.
WARN:  Expected /Users/curtis/dev/blocksworld-play-site/_build/dev/lib/certifi to be an app dir (containing ebin/*.app), but no .app found.
==> hackney (compile)
WARN:  Expected /Users/curtis/dev/blocksworld-play-site/_build/dev/lib/mimerl to be an app dir (containing ebin/*.app), but no .app found.
WARN:  Expected /Users/curtis/dev/blocksworld-play-site/_build/dev/lib/certifi to be an app dir (containing ebin/*.app), but no .app found.
Dependency not available: mimerl-.* ({git,
                                      "https://github.com/benoitc/mimerl",
                                      {tag,"1.0.0"}})
Dependency not available: certifi-.* ({git,
                                       "https://github.com/certifi/erlang-certifi",
                                       {tag,"0.3.0"}})
ERROR: compile failed while processing /Users/curtis/dev/blocksworld-play-site/deps/hackney: rebar_abort
** (Mix) Could not compile dependency :hackney, "/Users/curtis/.mix/rebar" command failed. You can recompile this dependency with "mix deps.compile hackney", update it with "mix deps.update hackney" or clean it with "mix deps.clean hackney"

**Comment by benoitc** [comment]
you probably need to update the registry.

**Comment by curtislinden** [comment]
what is the registry?

**Comment by ericmj** [comment]
The registry is always automatically updated with mix.

Using master won't test if this works because it only breaks when using mix and hex, using git master doesn't use hex of course.

I cannot reproduce @curtislinden errors, but I get other errors when using mix and hackney master, the issue is this line: https://github.com/benoitc/hackney/blob/master/rebar.config.script#L7. Mix doesn't support rebar3 so it should return `false`. Rebar3 is expected to land in elixir 1.2.0, so you can do a version check like you do above but check for mix version 1.2.0.

**Comment by curtislinden** [comment]
reverted to hackney 1.3.2 for the moment.

@ericmj  thx for the input.

**Comment by benoitc** [comment]
@ericmj well how does mix get the dependencies? Is there a way to tell it to not use git but hex packages?

**Comment by curtislinden** [comment]
awesome - working now - thxn!


On Tue, Nov 24, 2015 at 11:49 AM, Benoit Chesneau <notifications@github.com>
wrote:

> Closed #261 <https://github.com/benoitc/hackney/issues/261> via c298aed
> <https://github.com/benoitc/hackney/commit/c298aede85f6ff3af7174b4e12aa5d6e007541b1>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/benoitc/hackney/issues/261#event-473716087>.
>



-- 
Curtis J Schofield | Senior Systems Analyst | U2U
Skype robotarmymade | Second Life Curtis Linden | C 415.632.6001

Linden Lab | Makers of Shared Creative Spaces
